src/finite_element/finite_element_field_evaluation.cpp

```diff
@@ -474,7 +474,7 @@ int FE_element_field_evaluation::calculate_values(FE_field *fieldIn,
 							}
 							// transform values to standard basis functions if needed
 							FE_basis *basis = eft->getBasis();
```

Comment:
You are redeclaring this here, is that what you meant to do? Further down you test to see if basis is the same as the current eft->basis() but it will always be the nullptr I think.

Reply:
I'll make it clearer and maintain 'lastBasis' between iterations, which is initially nullptr so basis of first component is set up and subsequent components reuse last basis mappings until it changes.